### PR TITLE
Fix latest batch of user errors

### PR DESCRIPTION
**Pull request by jbygdell**
* Converts windows pat to unix when using the `-targetDir` option, tests on windows by using `filepath.join()`
* sets correct MultipartChunkSizeMb if set below 15MB
* Fixes some linter issues
* print to stdout instead of log when listing uploads

Fixes: #188 #187

### CODE REVIEW

File: upload/upload_test.go [hunk 1]
```diff
@@ -361,11 +355,11 @@ func (suite *TestSuite) TestFunctionality() {
 
 	// Test recursive upload
 	os.Args = []string{"upload", "-config", configPath.Name(), "-r", dir}
-	err = Upload(os.Args)
-	assert.NoError(suite.T(), err)
+	assert.NoError(suite.T(), Upload(os.Args))
 
 	// Check logs that file was uploaded
 	logMsg := fmt.Sprintf("%v", strings.TrimSuffix(str.String(), "\n"))
+	// fmt.Printf("FOO: %s\n", logMsg)
```
Comment (norling) [inline upload/upload_test.go:362]: probably forgot this line here
```suggestion
```

File: upload/upload_test.go [hunk 2]
```diff
@@ -379,13 +373,14 @@ func (suite *TestSuite) TestFunctionality() {
 	assert.Equal(suite.T(), aws.StringValue(result.Contents[0].Key), fmt.Sprintf("%s/%s", filepath.Base(dir), filepath.Base(testfile.Name())))
 
 	// Test upload to a different folder
-	os.Args = []string{"upload", "-config", configPath.Name(), testfile.Name(), "-targetDir", "a"}
-	err = Upload(os.Args)
-	assert.NoError(suite.T(), err)
-
+	targetPath := filepath.Join("a", "b", "c")
+	// fmt.Printf("target path: %s\n", targetPath)
```
Comment (norling) [inline upload/upload_test.go:377]: more old debug lines
```suggestion
```

File: upload/upload_test.go [hunk 3]
```diff
@@ -379,13 +373,14 @@ func (suite *TestSuite) TestFunctionality() {
...
+	os.Args = []string{"upload", "-config", configPath.Name(), testfile.Name(), "-targetDir", targetPath}
+	assert.NoError(suite.T(), Upload(os.Args))
+	// fmt.Printf("\n\nBAR: %v\n", strings.TrimSuffix(str.String(), "\n"))
```
Comment (norling) [inline upload/upload_test.go:380]:
```suggestion
```

File: upload/upload_test.go [hunk 4]
```diff
@@ -395,7 +390,8 @@ func (suite *TestSuite) TestFunctionality() {
 	if err != nil {
 		log.Panic(err.Error())
 	}
-	assert.Equal(suite.T(), aws.StringValue(result.Contents[0].Key), fmt.Sprintf("a/%s", filepath.Base(testfile.Name())))
+	// fmt.Printf("%s/%s\n", filepath.ToSlash(targetPath), filepath.Base(testfile.Name()))
```
Comment (norling) [inline upload/upload_test.go:393]:
```suggestion
```

**Review by norling — APPROVED** [review]
some forgotten debug printouts, otherwise fine as far as I can tell! :)

File: upload/upload_test.go [hunk 5]
```diff
@@ -471,4 +458,103 @@ func (suite *TestSuite) TestFunctionality() {
...
+	assert.Contains(suite.T(), logMsg, msg)
+
+	// Check that file showed up in the s3 bucket correctly
+	result, err := s3Client.ListObjects(&s3.ListObjectsInput{
+		Bucket: aws.String("dummy"),
+	})
+	if err != nil {
+		log.Printf(err.Error())
+	}
+	assert.Equal(suite.T(), filepath.ToSlash(filepath.Join(targetPath, filepath.Base(dir), filepath.Base(testfile.Name()))), aws.StringValue(result.Contents[0].Key))
```
Comment (aaperis) [inline upload/upload_test.go:557]: No need for a new function, you can add this part before line 392 and then increase the index in line 428 to 2 for that test to pass.

**Review by aaperis — APPROVED** [review]
There is some code repetition in the added test which I think can be avoided, but otherwise it looks good :-)

File: upload/upload_test.go (hunk 5 above)
Comment (jbygdell) [reply upload/upload_test.go:557]: The whole point of having the tests independent is to track failures without having the cascading effects of testing everything together.

**Pull request merged**